apple/swift-docc
# allow arbitrary platform strings in the `@Available` directive

Bug/issue #, if applicable: rdar://104224866

## Summary

In https://github.com/apple/swift-docc/pull/440, the `@Available` directive was limited to four platforms: `iOS`, `macOS`, `watchOS`, `tvOS`. This PR opens up that directive to take arbitrary platform strings, which will be passed along to the rendered page.

## Dependencies

None

## Testing

Use the following metadata for testing:

```markdown
@Metadata {
    @Available("Swift DocC", introduced: "1.0")
}
```

Steps:
1. Add the above metadata to `Sources/SwiftDocC/SwiftDocC.docc/SwiftDocC/Directives.md`.
2. `bin/preview-docs`
3. Ensure that the "Directives" page is printed with an availability callout for "Swift DocC" (with the space - i'm specifically testing platforms with spaces here):

<img width="698" alt="image" src="https://user-images.githubusercontent.com/5217170/212437448-6af0bda9-e94c-4305-bc0e-9d77a8345f73.png">

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
